tfx/components/example_gen/import_example_gen/executor.py

```diff
@@ -43,12 +44,17 @@ def _ImportSerializedRecord(  # pylint: disable=invalid-name
       that maps to input files with root directory given by input_base.
 
   Returns:
-    PCollection of records (tf.Example, tf.SequenceExample, or bytes).
+    PCollection of records (tf.Example, tf.SequenceExample, bytes or dictionaries).
   """
   input_base_uri = exec_properties[standard_component_specs.INPUT_BASE_KEY]
   input_split_pattern = os.path.join(input_base_uri, split_pattern)
   logging.info('Reading input TFRecord data %s.', input_split_pattern)
 
+  if output_payload_format == example_gen_pb2.PayloadFormat.FORMAT_PARQUET:
+    return (pipeline
+            | "ReadParquet" >>
+            beam.io.ReadFromParquet(file_pattern=input_split_pattern))
```

Comment:
ReadFromParquet returns PCollection of dict? if we convert it to bytes here, we can save one conversion in base_executor & write_split